client/engine/store/memstore.go

```diff
@@ -245,6 +264,26 @@ func (ms *MemStore) GetConsensusChannel(counterparty types.Address) (channel *co
 	return
 }
 
+func (ms *MemStore) GetAllConsensusChannels() ([]*consensus_channel.ConsensusChannel, error) {
+	toReturn := []*consensus_channel.ConsensusChannel{}
+	var err error
+	ms.consensusChannels.Range(func(key string, chJSON []byte) bool {
+		var ch consensus_channel.ConsensusChannel
+
+		err = json.Unmarshal(chJSON, &ch)
+		if err != nil {
+			return false
+		}
```

Comment:
This seems like a better approach: stop iterating and then bubble the error up. Is there a reason we can't do that in all cases and in all our store implementations?

Reply:
I think this makes sense to apply to other implementations. For the purposes of this PR I'll update the the functions added.